Feature/lógica calculadora

PatrickPSouza opened the pull request:
--alteração na lógica da função calcular

-- criação e alteração de testes unitários

PatrickPSouza committed:
chore: mudar formulas do cálculo

Mudar a forma como o cálculo é feito, passando de simples para composto, adicionando ROI e Montante.

Mudar o tipo de dados do tempo, de DIA para MESES.

PatrickPSouza committed:
feat: Funcionalidade de validar os valores de entrada

Cria uma validação para verificar se os valores de entrada são do tipo number.

PatrickPSouza committed:
test: adiciona novos testes unitários e edita existentes

Adiciona teste unitário de funcionamento recebendo caracteres especiais.
altera o expect do teste de cálculo com valores negativos.
edita a lógica de funcionamento com dois valores válidos e um valor inválido

PatrickPSouza committed:
fix: corrigir validação de dados na função

Corrige o trecho de código que verifica se todos os dados informados são do tipo number, e lança um erro caso não sejam.

o arquivo de tests foi comentado, para ser corrigido

PatrickPSouza committed:
feat: adicionar validação de números negativos

adiciona uma validação que verifica se os valores são negativos. Caso sejam, lança um erro.

PatrickPSouza committed:
test: faz os test cases propóstos no arquivo MD

Realiza três testes de suite cases:
### 1 - Testar o funcionamento da função calcularDividendo de acordo com os inputs no campo valor
### 2 - Testar o funcionamento da função calcularDividendo de acordo com os inputs no campo tempo
### 3 - Testar o cálculo da função calcularDividendo.

Sownatta commented on build/src/Factory.js:
const/let poderia ser uma opção melhor para as variáveis declaradas como var (?)

PatrickPSouza replied:
Conforme conversamos. Quando é feita a Build do JS para o TS, por padrão, as variáveis ficam como var.

Sownatta commented on build/src/services/calcular.js:
Remover comentários desnecessários da linha 25 à 38.

PatrickPSouza replied:
Removidos

Sownatta commented on src/services/calcular.ts:
Remover comentários desnecessários da linha 29 à 41.

PatrickPSouza replied:
removidos

PatrickPSouza replied:
Removidos

Sownatta commented on src/tests/calcular.spec.ts:
Remover comentários da linha 157 à 212.

PatrickPSouza replied:
arquivo de teste refeito

Sownatta reviewed (commented):
- Revisar variáveis declaradas com var; 
- Remover comentários desnecessários nos arquivos indicados nos comentários;
- O arquivo app.js parece estar duplicado e em diferentes versões dentro de /build;
- Não há conflitos para resolver.

PatrickPSouza committed:
refactor: remove comentarios e arquivo extra

Remove comentários desnecessários no arquivo calcular.js e exclui arquivo extra de app.js

Sownatta reviewed (approved):
Tudo nos conformes.
